assets/js/data/cart/selectors.js

```diff
@@ -106,3 +106,13 @@ export const isRemovingCoupon = ( state ) => {
 export const getCouponBeingRemoved = ( state ) => {
 	return state.metaData.removingCoupon || '';
 };
+
+/**
+ * Retrieves the selected shipping rates ids.
+ *
+ * @param {Object} state The current state.
+ * @return {string} The data to return.
+ */
+export const selectedShippingRates = ( state ) => {
+	return state.cartData.selectedShippingRates;
```

Comment:
so this works fine, but if I dispatch this no longer works and all keys are now snake_case.

Reply:
So it looks like the other place `receiveCart` is being used automatically handles snake-casing the keys: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/abfa541dc9398519a845a17a9581bcfb1e2005a1/assets/js/data/cart/resolvers.js#L28-L33

So what could be done is doing this key conversion in the action itself otherwise this is going to happen anytime the action creator is called.

Reply:
what confuses me is that apply coupon and removeCoupn do not call it. I will test this and see

Reply:
Huh! You're right. Might be a bug unless the api is returning a different shape for coupon requests - which if so, probably isn't a good idea anyways.

So, here's what I suggest then (that differs a bit from my original suggestion). Since this is a state shaped/maintained by the reducer, the key normalization should happen in the reducer for any actions updating the `cartData` property. That way we're assured the state is always in the expected shape.